src/internal/observable/combineLatest.ts

```diff
@@ -522,44 +504,61 @@ export function combineLatestInit(
   return (subscriber: Subscriber<any>) => {
     // The outer subscription. We're capturing this in a function
     // because we may have to schedule it.
-    const primarySubscribe = () => {
-      const { length } = observables;
-      // A store for the values each observable has emitted so far. We match observable to value on index.
-      const values = new Array(length);
-      // The number of currently active subscriptions, as they complete, we decrement this number to see if
-      // we are all done combining values, so we can complete the result.
-      let active = length;
-      // A temporary array to help figure out if we have gotten at least one value from each observable.
-      const hasValues = observables.map(() => false);
-      let waitingForFirstValues = true;
-      // Called when we're ready to emit a set of values. Note that we copy the values array to prevent mutation.
-      const emit = () => subscriber.next(valueTransform(values.slice()));
-      // The loop to kick off subscription. We're keying everything on index `i` to relate the observables passed
-      // in to the slot in the output array or the key in the array of keys in the output dictionary.
-      for (let i = 0; i < length; i++) {
-        const subscribe = () => {
-          const source = from(observables[i] as ObservableInput<any>, scheduler as any);
-          source.subscribe(
-            new CombineLatestSubscriber(
-              subscriber,
-              (value) => {
-                values[i] = value;
-                if (waitingForFirstValues) {
-                  hasValues[i] = true;
-                  waitingForFirstValues = !hasValues.every(identity);
-                }
-                if (!waitingForFirstValues) {
-                  emit();
-                }
-              },
-              () => --active === 0
-            )
+    maybeSchedule(
+      scheduler,
+      () => {
+        const { length } = observables;
+        // A store for the values each observable has emitted so far. We match observable to value on index.
+        const values = new Array(length);
+        // The number of currently active subscriptions, as they complete, we decrement this number to see if
+        // we are all done combining values, so we can complete the result.
+        let active = length;
+        // The number of inner sources that still haven't emitted the first value
+        // We need to track this because all sources need to emit one value in order
+        // to start emitting values.
+        let remainingFirstValues = length;
```

Comment:
Here I'm using a countdown, instead of an array, to track whether or not each source has emitted at least one value.